src/Symfony/Component/Dsn/ConnectionFactory.php

```diff
@@ -0,0 +1,67 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Component\Dsn;
+
+use Symfony\Component\Dsn\Exception\InvalidArgumentException;
+use Symfony\Component\Dsn\Factory\MemcachedConnectionFactory;
+use Symfony\Component\Dsn\Factory\RedisConnectionFactory;
+
+/**
+ * Factory for undetermined Dsn.
+ *
+ * @author Jérémy Derussé <jeremy@derusse.com>
+ */
+class ConnectionFactory
```

Comment:
please make it final

Reply:
:-1: for having everything static here

Reply:
although we don't usually make things final by default, I think we should here, because there is no "type" to inherite from, only behavior to composer.
So :+1: for `final`.

Reply:
:-1: for non-static, there is no OOP here, just constructors.